src/lib/datamodel/ClusterOnOff.h

```diff
@@ -62,6 +70,44 @@ class ClusterOnOff : public Cluster
         AddAttribute(&mOnTime);
         AddAttribute(&mOffWaitTime);
     }
+
+    virtual CHIP_ERROR HandleCommandOff(const Command & cmd)
```

Comment:
why all this breakout in the base class?  why take a Command parameter if it goes unused?

Reply:
So my thought process here is that Clusters will have multiple commands: some of them can be handled within the Cluster itself (like in this case), but for some, the application will have to provide the handler.
 
Then, from an overall consistency perspective, if there is a cluster-specific command, the application can be handle it through `HandleCommand<cmdName>(cmd)`. Applications provide these implementations either to override the default implementation, or implement commands that can't be handled within the SDK anyway.

Thoughts?

Reply:
I think the only thing that the base clusters should be providing are those methods required for the base clusters to interact with each other.  So: on-off needs to export a vtable that works for things that level-control needs from on-off.   

This level of decomposition of the command in the base class looks a little like overkill.  Are device manufacturers  really looking for this many callbacks?

Reply:
It seems like an overkill, particularly in the onoff scenario here and could be eliminated. As we move to other scenarios, would it seem that way, or would most commands will need to be handled in the applications anyway?

Reply:
Currently, applications provide their functionality by overriding the function, thus adding to the vtable. I think what you are suggesting is to limit the overriding to cluster layering functionality, while for applications they can have their function callbacks that don't rely on a vtable?

Reply:
We only need vtable entries where code in other clusters needs to understand the clusterid<->interface mapping.  To override behavior in an application, those functions can be overridden and dispatched to whatever app code, while dispatching to the superclass for default behavior?